osf/migrations/0171_add_registration_files_count.py

```diff
@@ -2,40 +2,38 @@
 # Generated by Django 1.11.15 on 2019-01-18 14:43
 from __future__ import unicode_literals
 import logging
-from tqdm import tqdm
 
 from django.db import migrations, models
-from django.db.models import Count
 from django_bulk_update.helper import bulk_update
-from osf.models import Registration
 
 logger = logging.getLogger(__file__)
 
 def add_registration_files_count(state, *args, **kwargs):
     """
     Caches registration files count on Registration object.
-
     Importing Registration model outside of this method to take advantage of files
     relationship for speed purposes in this migration.  If this model changes significantly,
     this migration may have to be modified in the future so it runs on an empty db.
     """
-    registrations = Registration.objects.filter(is_deleted=False).filter(
-        files__type='osf.osfstoragefile',
-        files__deleted_on__isnull=True
-    ).annotate(
-        annotated_file_count=Count('files')
-    )
-    progress_bar = tqdm(total=registrations.count())
+    Registration = state.get_model('osf', 'Registration')
+    registrations = Registration.objects.filter(is_deleted=False, files_count__isnull=True)
+    BaseFileNode = state.get_model('osf', 'BaseFileNode')
+    ContentType = state.get_model('contenttypes', 'ContentType')
+    content_type = ContentType.objects.get(app_label='osf', model='registration')
     registrations_to_update = []
 
-    for i, registration in enumerate(registrations, 1):
-        progress_bar.update(i)
-        registration.files_count = registration.annotated_file_count
+    for registration in registrations:
+        registration_files = BaseFileNode.objects.filter(
+            target_object_id=registration.id,
+            target_content_type=content_type,
+            type='osf.osfstoragefile',
+            deleted_on__isnull=True,
+        )
+        registration.files_count = registration_files.count()
         registrations_to_update.append(registration)
 
     bulk_update(registrations_to_update, update_fields=['files_count'], batch_size=5000)
     logger.info('Populated `files_count` on a total of {} registrations'.format(len(registrations_to_update)))
-    progress_bar.close()
```

Comment:
AbstractNode model has changed in future migrations, which breaks this migration which imported the Registration model outside of this migration - borrowing Corbin's changes in https://github.com/CenterForOpenScience/osf.io/pull/9133